test/unit/services/auth_service_test.dart

```diff
@@ -147,5 +156,125 @@ void main() {
             emitsDone,
           ]));
     });
+
+    test(
+        'emailSignInStream emits UpdateEmailAuthOptionsPage actions at each stage of successful sign in',
+        () {
+      final service = AuthService(FakeFirebaseAuth1(), null, null);
+
+      expect(
+          service.emailSignInStream('test@email.com', 'test_password'),
+          emitsInOrder(<dynamic>[
+            UpdateEmailAuthOptionsPage(step: AuthStep.signingInWithEmail),
+            UpdateEmailAuthOptionsPage(step: AuthStep.waitingForInput),
+            RemoveCurrentPage(),
+            emitsDone,
+          ]));
+    });
+
+    test(
+        'emailSignInStream catches firebaseAuthExceptions and emits StoreProblem actions',
+        () async {
+      // init service to throw firebaseAuthException on sign in
+      final service =
+          AuthService(FakeFirebaseAuthSignInException(), null, null);
+
+      /// Check that authService emits [AddProblem] action with info
+      /// from caught [firebaseAuthException].
+      /// We use a [TypeMatcher] as it's difficult to create the expected
+      /// [Problem] due to the [Problem.trace] member
+      expect(
+          service.emailSignInStream('test@email.com', 'test_password'),
+          emitsInOrder(<dynamic>[
+            UpdateEmailAuthOptionsPage(step: AuthStep.signingInWithEmail),
+            UpdateEmailAuthOptionsPage(step: AuthStep.waitingForInput),
+            TypeMatcher<AddProblem>()
+              ..having((p) => p.problem.type, 'type', ProblemType.emailSignIn)
+              ..having(
+                  (p) => p.problem.info,
+                  'info',
+                  BuiltMap<String, Object>(
+                      {'code': 'test error: cant find user'}))
+              ..having((p) => p.problem.message, 'message',
+                  equals('firebase auth exception error')),
+            emitsDone
+          ]));
+    });
+
+    test('emailSignInStream catches errors and emits StoreProblem actions',
+        () async {
+      // Init service that throws exception when signing in with firebase
+      final service = AuthService(FakeFirebaseAuthThrows(), null, null);
+
+      /// Check that emailSignInStream emits addProblem action with info from caught error.
```

Comment:
addProblem -> AddProblem (seems like a super nit but the meaning the case conveys here is worth caring about I reckon)